packages/app-admin-ui/index.js

```diff
@@ -160,7 +160,13 @@ class AdminUIApp {
         default: () => {
           next();
         },
+        // Without this, request with an 'Accept: */*' header get picked up by
+        // the 'text/html' handler rather than the default
         '*/*': () => {
+          // We need to reset the res 'Content-Type' otherwise it gets replaced by the format we've matched on: '*/*'.
+          // Returning a wildcard mimetype causes problems if a 'X-Content-Type-Options: nosniff' header is also set.
+          // See.. https://github.com/keystonejs/keystone/issues/2741
+          res.type(path.extname(req.url));
```

Comment:
What happens when `req.url` = `/foo/bar`?

Wouldn't we want the content type to be `text/html` then? Or maybe `*/*` to let the client decide?

Reply:
Don't we control 100% of the requests? In the adminUI... which so far I can only see `/admin/js/*`

Reply:
Maybe something like this?
```suggestion
          const extension = path.extname(req.url);
          res.type(extension ? extension : '*/*');
```

Reply:
@dominikwilkowski This express app handles _everything_ under `/admin`, which could include custom pages. Not to mention the pages such as `/admin/todos` which doesn't have an extension.

Reply:
Great point. So...

1. If I request `/admin/blerg` from the browser (address bar), the browser (Chrome in this case) adds a sensible `Accept` header of..

```
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9
```

Which works fine as it matches the `text/html` resolver (returns with `Content-Type: text/html; charset=utf-8`).

2. If I request `/admin/blerg` _without_ an `Accept` header, it hits the `*/*` format resolve and the `Content-Type` is returned as `application/octet-stream`. Presumably this is some kind of default that's applied since we're passing `res.type()` and empty string.

3. If I request with an unrecognised `Accept` header (`food/pasta`) it hits the `default` format resolver, returning with `Content-Type: text/html; charset=UTF-8` 
which is correct for the content -- the main html page for the admin UI.

This all seems to be true regardless of the depth of the path, trailing slashes, etc.

Point 2 is probably wrong. The content being returned is actual html; the incorrect mime type here is our fault -- I've patched to only override if a file extension is present, meaning it will return `*/*` as before.

Note this still isn't totally correct. If we removed the `res.format()` block entirely, these requests would return `text/html`. But as you say, at least the browser can decide (and it's still better than we have now).